webpack/components/extensions/HostDetails/Cards/ContentViewDetailsCard/ChangeHostCVModal.js

```diff
@@ -0,0 +1,258 @@
+  );
+};
+
+ContentViewDescription.propTypes = {
+  cv: PropTypes.shape({
+    default: PropTypes.bool.isRequired,
+    id: PropTypes.number.isRequired,
+    latest_version: PropTypes.string.isRequired,
+  }).isRequired,
+  versionNumber: PropTypes.string.isRequired,
+};
+
+const ChangeHostCVModal = ({
+  isOpen,
+  closeModal,
+  hostEnvId,
+  orgId,
+  hostId,
+  hostName,
+}) => {
+  const [selectedEnvForHost, setSelectedEnvForHost]
+    = useState([]);
+
+  const [selectedCVForHost, setSelectedCVForHost] = useState(null);
+  const [cvSelectOpen, setCVSelectOpen] = useState(false);
+  const dispatch = useDispatch();
+  const contentViewsInEnvResponse = useSelector(state => selectContentViews(state, `FOR_ENV_${hostEnvId}`));
+  const contentViewsInEnvStatus = useSelector(state => selectContentViewStatus(state, `FOR_ENV_${hostEnvId}`));
+  const hostUpdateStatus = useSelector(state => selectAPIStatus(state, HOST_CV_AND_ENV_KEY));
+
+  useAPI( // No TableWrapper here, so we can useAPI from Foreman
+    'get',
+    api.getApiUrl(`/organizations/${orgId}/environments/paths?permission_type=promotable`),
+    ENV_PATH_OPTIONS,
+  );
+
+  const handleModalClose = () => {
+    setCVSelectOpen(false);
+    setSelectedCVForHost(null);
+    setSelectedEnvForHost([]);
+    closeModal();
+  };
+
+  const selectedEnv = selectedEnvForHost?.[0];
+  const selectedEnvId = selectedEnv?.id;
+
+  const handleCVSelect = (event, selection) => {
+    setSelectedCVForHost(selection);
+    setCVSelectOpen(false);
+  };
+
+  const handleEnvSelect = (selection) => {
+    dispatch(getContentViews({
+      environment_id: selection[0].id,
+      include_default: true,
+      full_result: true,
+      order: 'default DESC', // show Default Organization View first
+    }, `FOR_ENV_${hostEnvId}`));
+    setSelectedCVForHost(null);
+    setSelectedEnvForHost(selection);
+  };
+  const { results: contentViewsInEnv = [] } = contentViewsInEnvResponse;
+  const canSave = !!(selectedCVForHost && selectedEnvForHost.length);
+
+  const relevantVersionObjFromCv = (cv, env) => { // returns the entire version object
+    const versions = cv.versions.filter(version => new Set(version.environment_ids).has(env.id));
+    return uniq(versions)?.[0];
+  };
+  const relevantVersionFromCv = (cv, env) =>
+    relevantVersionObjFromCv(cv, env)?.version; // returns the version text e.g. "1.0"
+
+  const refreshHostDetails = () => {
+    handleModalClose();
+    return dispatch({
+      type: 'API_GET',
+      payload: {
+        key: HOST_DETAILS_KEY,
+        url: `/api/hosts/${hostName}`,
+      },
+    });
+  };
+
+  const handleSave = () => {
+    const requestBody = {
+      id: hostId,
+      host: {
+        content_facet_attributes: {
+          content_view_id: selectedCVForHost,
+          lifecycle_environment_id: selectedEnvId,
+        },
+      },
+    };
+    dispatch(updateHostContentViewAndEnvironment(
+      requestBody, hostId,
+      refreshHostDetails, handleModalClose,
```

Comment:
I noticed that when there is an error, the modal didn't close, making it hard to see the error toast. So I passed `handleModalClose` here as a `handleError`.